python/tests/util/storage_test.py

```diff
@@ -0,0 +1,241 @@
+    - value contains integers in ascending order
+    - fake_ticker contains strings that are generated based on the corresponding value with the gen_fake_ticker function
+    """
+    # Generate random values such that their sum is equal to n
+    values = range(1, n + 1)
+    # Generate timestamps
+    timestamps = pd.date_range(end=end_timestamp, periods=n, freq=freq)
+    # Generate timestamps
+    fake_tickers = [gen_fake_ticker(val) for val in values]
+    # Create dataframe
+    df = pd.DataFrame({"timestamp": timestamps, "fake_ticker": fake_tickers, "value": values})
+    return df
+
+
+def check_timestamps(df):
+    # Calculate differences between each timestamp and its previous timestamp
+    differences = df["timestamp"].diff().dt.total_seconds()
+    # Check if all differences are equal to 1 second
+    all_one_second = (differences[1:] == 1).all()
+    return all_one_second
+
+
+def check_fake_tickers(df):
+    # Generate expected fake_tickers from 'values' column
+    expected_fake_tickers = [gen_fake_ticker(val) for val in df["value"]]
+    # Create a mask that indicates which 'fake_tickers' are not equal to the expected ones
+    mask = df["fake_ticker"] != expected_fake_tickers
+    # If there are any unexpected tickers, print them
+    if mask.any():
+        print("Actual values:")
+        print(df[mask])
+        print("Expected values:")
+        print([expected_fake_tickers[i] for i in range(len(mask)) if mask[i]])
+    # Check if all 'fake_tickers' are equal to expected_fake_tickers
+    is_correct = not mask.any()
+    return is_correct
+
+
+def verify_ascending_dataframe(df, n):
+    """
+    Helper function that is used to verify a Data Frame that was generated by generate_ascending_dataframe
+    """
+    value_sum = df.value.sum()
+    assert n == len(df)
+    assert value_sum == ((n * (n + 1)) / 2)
+    assert check_timestamps(df)
+    assert check_fake_tickers(df)
+    assert check_last_timestamp(df)
+
+
+def verify_pseudo_random_dataframe(df, n):
+    """
+    Helper function that is used to verify a Data Frame that was generated by generate_pseudo_random_dataframe
+    """
+    value_sum = df.value.sum()
+    assert n == len(df)
+    assert n - 1 < value_sum < n + 1
+    assert check_last_timestamp(df)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        prog="storage_test",
+        description="wrapper for the functionalities that are needed for persistent storage testing",
+    )
+
+    parser.add_argument("-t", "--type", required=True)
+    parser.add_argument(
+        "-v",
+        "--version",
+        required=False,
+        help=(
+            "The version of ArcticDB that is used for seed, this is to make the name of the library that is seeded"
+            " unique"
+        ),
+    )
+    args = parser.parse_args()
+    job_type = str(args.type).lower()
+    # TODO: Add support for other storages
```

Comment:
Can you make a github ticket to do Azure?

Reply:
Done - issue #876